lib/delayed/backend/active_record.rb

```diff
@@ -68,12 +68,17 @@ def self.set_delayed_job_table_name
         set_delayed_job_table_name
 
         def self.ready_to_run(worker_name, max_run_time)
-          where(
-            "((run_at <= ? AND (locked_at IS NULL OR locked_at < ?)) OR locked_by = ?) AND failed_at IS NULL",
```

Comment:
i think those parens actually need to be there... the locking logic has something to do with being able to find jobs where are overdue and already locked (and eligible to be re-picked up) or that they were locked by the current container.

Reply:
WTH? I didn't edit those. Must have been rufo formatting code that it thought was safe.